server/pulp/server/content/web/views.py

```diff
@@ -51,52 +53,91 @@ def urljoin(scheme, host, port, base, path, query):
+
     def __init__(self, **kwargs):
-        super(RedirectView, self).__init__(**kwargs)
+        super(ContentView, self).__init__(**kwargs)
         self.key = Key.load(pulp_conf.get('authentication', 'rsa_key'))
-        self.alias = AliasTable()
-        self.alias.load()
 
-    def get(self, request):
+    def redirect(self, request, path):
         """
         Redirected GET request.
 
         :param request: The WSGI request object.
         :type request: django.core.handlers.wsgi.WSGIRequest
-        :return: A redirect or not-found response.
+        :param path: The *real* path to stored content.
+        :type path: str
+        :return: A redirect or not-found reply.
         :rtype: django.http.HttpResponse
         """
-        try:
-            path = request.environ['REDIRECT_URL']
-        except KeyError:
-            # Not redirected by ErrorDocument
-            return HttpResponseNotFound()
+        enabled = parse_bool(pulp_conf.get('lazy', 'enabled'))
+
+        if not enabled:
+            # Not Redirected
+            return HttpResponseNotFound(path)
 
-        query = request.environ.get('REDIRECT_QUERY_STRING', '')
         scheme = request.environ['REQUEST_SCHEME']
         host = request.environ['SERVER_NAME']
         port = request.environ['SERVER_PORT']
-        enabled = parse_bool(pulp_conf.get('lazy', 'enabled'))
+        query = request.environ.get('REDIRECT_QUERY_STRING', '')
+
         redirect_host = pulp_conf.get('lazy', 'redirect_host')
         redirect_port = pulp_conf.get('lazy', 'redirect_port')
         redirect_path = pulp_conf.get('lazy', 'redirect_path')
-        if enabled:
-            redirect = self.urljoin(
-                scheme,
-                redirect_host or host,
-                redirect_port or port,
-                redirect_path,
-                self.alias.translate(path),
-                query)
-            url = URL(redirect)
-            signed = url.sign(self.key)
-            return HttpResponseRedirect(str(signed))
+
+        redirect = self.urljoin(
+            scheme,
+            redirect_host or host,
+            redirect_port or port,
+            redirect_path,
+            path,
+            query)
+
+        url = URL(redirect)
+        signed = url.sign(self.key)
+        return HttpResponseRedirect(str(signed))
+
+    def get(self, request):
+        """
+        Process the GET content request.
+
+        :param request: The WSGI request object.
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :return: An appropriate HTTP reply
+        :rtype: django.http.HttpResponse
+        """
+        host = request.get_host()
+        path = os.path.realpath(request.path)
+
+        # Authorization
+        if not allow_access(request.environ, host):
+            # Not Authorized
+            return HttpResponseForbidden()
+
+        # Delivery
+        if os.path.exists(path):
+            return self.x_send(path)
         else:
```

Comment:
Completely personal preference (and very quibbly), but I feel like it would read a bit more smoothly if the check to see if lazy is enabled happened here rather than in the `redirect` method, so something like 
```
if path exists:
    send file
elif lazy not enabled:
    404
else:
    redirect
```
and then remove the enabled check from redirect.